### DISCUSSION TITLE
How to create two rows at the same time by grouping two specific values as a set

### DISCUSSION THREAD

**Discussion by gyoeun666** [community/community#35227] [Other Feature Feedback, Questions, & Ideas]
I have a data_type table and a data table. The column structure is as follows.
```
data_type    data
=========    ====================
id | type    type_ID | count
---------    --------------------
 1 | a             1 | 50
 2 | b             2 | 100
 3 | c             3 | 30
```
In data_type, types 'a' and 'b' are sets, and they must be added to the data table at the same time, and only one of them cannot be entered.

For example
```
INSERT INTO data (typeId, count)
VALUES (1,50), (2,90) <- O

INSERT INTO data (typeId, count)
VALUES (1,50), (3,100) <- X
INSERT INTO data (typeId, count)
VALUES (1,50) <- X
```
Is there any way to do this with SQL rather than ORM?
I need help.

Below is my current query.
DAO.js
```
const createData = async(userId, a1, a2, typeId, count) => {
  await myDataSource.query(
    `INSERT INTO a (user_id, a1, a2)
  VALUES(?,?,?)`,
    [user_id, a1, a2]
  );

  const typeAndCount = typeId
    .map((type, index) => `((SELECT LAST_INSERT_ID()),${type},${count[index]})`)
    .join(",");

  const data = await myDataSource.query(
    `INSERT INTO data (a_id ,type_id, count)
  VALUES ${typeAndCount}`
  );
  return data;
};
```

**Comment by airtower-luna** [comment]
All SQL implementations I know about support [transactions](https://en.wikipedia.org/wiki/Database_transaction). Highly simplified, a transaction is a group of operations where either all or none of them are applied. Check the documentation of the API you're using for how to make transactions.
